lightning/src/ln/onion_utils.rs

```diff
@@ -935,6 +935,160 @@ pub(crate) fn decode_next_payment_hop<NS: Deref>(
 	}
 }
 
+/// Build a payment onion, returning the first hop msat and cltv values as well.
+pub fn create_payment_onion<T>(
+	secp_ctx: &Secp256k1<T>, path: &Path, session_priv: &SecretKey, total_msat: u64, 
+	recipient_onion: RecipientOnionFields, starting_htlc_offset: u32, payment_hash: PaymentHash,
```

Comment:
`starting_htlc_offset` should be named `best_block_height`, then we should add 1 before passing it into `build_onion_payloads` (this is what `ChannelManager / outbound_payment` does when sending an htlc).

Reply:
Got it, thanks